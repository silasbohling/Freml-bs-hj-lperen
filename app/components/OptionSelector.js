import React from 'react'
import { View, Text, StyleSheet, TouchableOpacity, Animated } from 'react-native'
import Colors from '@modules/Colors.js'
import Utils from '@modules/Utils'

export default class OptionSelector extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      offset: new Animated.Value(0)
    }

    this.width = Utils.wp(80)
  }

  _optionSelected = (selected, value) => {
    Animated.timing( this.state.offset, {
      toValue: this.width/this.props.options.length*selected,
      duration: 300
    }).start( () => {
      this.props.optionPressed(value)
    })
  }

  render() {
    const renderedOptions = this.props.options.map( (option, index) => {
      return(
        <TouchableOpacity
          style={ styles.option }
          onPress={ () => this._optionSelected(index, option.value) }
          key={index}>
          <Text style={styles.label}>{ option.label }</Text>
        </TouchableOpacity>
      )
    })

    return(
      <View style={[styles.container, { width: this.width }]}>
        <Animated.View
          style={[ styles.selected, { width: this.width/this.props.options.length, left: this.state.offset } ]}
        />
        {renderedOptions}
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.secondary,
    borderRadius: 10,
    flexDirection: 'row'
  },
  option: {
    borderRadius: 10,
    paddingVertical: 10,
    flex: 1,
    alignItems: 'center'
  },
  label: {
    color: Colors.btnText,
    fontSize: 12,
    fontWeight: '500'
  },
  selected: {
    backgroundColor: Colors.primary,
    position: 'absolute',
    top: 0,
    bottom: 0,
    borderRadius: 10,
  }
})
