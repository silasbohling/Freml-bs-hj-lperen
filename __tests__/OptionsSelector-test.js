import React from 'react';
import OptionSelector from '@components/OptionSelector';

import Renderer from 'react-test-renderer';

describe( 'OptionSelector', () => {

  it('render correctly', () => {
    const options = [
      { label:"24 timer", value:9 },
      { label:"48 timer", value:17 },
      { label:"3 døgn", value:25 },
      { label:"4 døgn", value:33 }
    ]

    const component = <OptionSelector options={ options } optionPressed={null} />
    const three = Renderer.create(component).toJSON()

    expect(three).toMatchSnapshot()
  })
})
