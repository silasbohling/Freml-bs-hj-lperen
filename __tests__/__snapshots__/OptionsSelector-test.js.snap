// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionSelector render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,0,0, 0.8)",
        "borderRadius": 10,
        "flexDirection": "row",
      },
      Object {
        "width": 600,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(183,32,33)",
        "borderRadius": 10,
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 150,
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flex": 1,
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      24 timer
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flex": 1,
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      48 timer
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flex": 1,
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      3 døgn
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flex": 1,
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      4 døgn
    </Text>
  </View>
</View>
`;
